core/store/src/trie/update.rs

```diff
@@ -167,12 +208,12 @@ impl TrieUpdate {
     }
 }
 
-struct MergeIter<'a, I: Iterator<Item = (&'a Vec<u8>, &'a Option<Vec<u8>>)>> {
-    left: Peekable<I>,
-    right: Peekable<I>,
+struct MergeIter<'a> {
+    left: Peekable<Box<dyn Iterator<Item = (&'a Vec<u8>, &'a Option<Vec<u8>>)> + 'a>>,
```

Comment:
It seems that we cannot name a type that is used for the committed changes manipulated with `.map()`, so I had to resort to dynamic dispatch.

Reply:
That's perfectly fine.

Reply:
You could've written `Peekable<I1>` and then `I1: Iterator<Item = (&'a Vec<u8>, &'a Option<Vec<u8>>)> + 'a`

Reply:
If static type is used like that, it poisons many other places with this type parameter, e.g. `TrieUpdateIterator` would require a type parameter as well since we cannot express a type of `Map<Iterator, [closure]>`.